File: py/404.py
class Solution:
    def sumOfLeftLeaves(self, root:Optional[TreeNode]) -> int:
        stk = [root]
        sumll = 0

        while stk:
            node = stk.pop()

            if node:
                if node.left and node.left.left is None and node.left.right is None:
                    sumll += node.left.val
                stk.append(node.right)
                stk.append(node.left)
        return sumll

File: go/404.go
func sumOfLeftLeaves(root *TreeNode) int {
    if root == nil {
        return 0
    }

    stack := []*TreeNode{root}

    leftLeavesSum := 0

    for len(stack) > 0 {
        curr := stack[len(stack)-1]
        stack = stack[:len(stack)-1]
        if curr.Right != nil {
            stack = append(stack, curr.Right)
        }
        if curr.Left != nil {
            if curr.Left.Left == nil && curr.Left.Right == nil {
                leftLeavesSum += curr.Left.Val
            }else {
                stack = append(stack, curr.Left)
            }
        }
    }
    return leftLeavesSum
}
